### PR TITLE
Languages generator refactoring

### PR DESCRIPTION
**Pull request by mshima**
Related to #10528 

- Move composing task to default priority, this improves workflow for blueprints.
- Centralize languages/i18n configuration into languages generator.
- Makes i18n workflow simpler.

---

Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/languages/index.js [hunk 1]
```diff
@@ -97,16 +105,14 @@ module.exports = class extends BaseBlueprintGenerator {
...
                     } else if (this.skipServer) {
-                        this.log(chalk.bold(`\nInstalling languages: ${this.languages.join(', ')} for client`));
+                        this.log(chalk.bold(`\nInstalling languages: ${this.languagesToApply.join(', ')} for client`));
                     } else {
-                        this.log(chalk.bold(`\nInstalling languages: ${this.languages.join(', ')}`));
+                        this.log(chalk.bold(`\nInstalling languages: ${this.languagesToApply.join(', ')}`));
                     }
-                    this.languagesToApply = this.languages || [];
                 } else {
                     this.log(chalk.bold('\nLanguages configuration is starting'));
```
Comment (pascalgrimaud) [inline generators/languages/index.js:117]: @mshima : should we remove this line ? As it appears when anwsering to the questions:

```
? Which *type* of application would you like to create? Monolithic application (recommended for simple projects)
? [Beta] Do you want to make it reactive with Spring WebFlux? No
? What is the base name of your application? burger

Languages configuration is starting <------------------------
? What is your default Java package name? io.github.pascalgrimaud.burger
? Do you want to use the JHipster Registry to configure, monitor and scale your application? No
? Which *type* of authentication would you like to use? JWT authentication (stateless, with a token)
? Which *type* of database would you like to use? SQL (H2, MySQL, MariaDB, PostgreSQL, Oracle, MSSQL)
? Which *production* database would you like to use? MySQL
? Which *development* database would you like to use? H2 with disk-based persistence
? Do you want to use the Spring cache abstraction? Yes, with the Ehcache implementation (local cache, for a single node)
? Do you want to use Hibernate 2nd level cache? Yes
? Would you like to use Maven or Gradle for building the backend? Maven
? Which other technologies would you like to use? 
? Which *Framework* would you like to use for the client? 
  Angular 
  React 
❯ Vue 
  No client 
```

**Comment by pascalgrimaud** [comment]
@mshima : I merged it too quickly. I found a bug when generating project with languages `en` + `fr`. Only `fr` files are generated:

```
    "nativeLanguage": "en",
    "packageFolder": "io/github/pascalgrimaud/burger",
    "jwtSecretKey": "ZmZhNjhhYjFlODRkNzZlMGFlN2FhZWRmNTkyZDdlMWZiMTlhZGI4ZWNkYTM2ZjVmMDI3N2I2NDVhMTAxZGVkM2JiNTYxYTk0YmFhNTQwYTQ0ZmY5Y2FlN2Q0Y2MzNTc5YmYwOWI3ZGM5MjJjMzM4YzRhN2M0OTU3ZTU1M2U0YWM=",
    "clientThemeVariant": "primary",
    "useSass": true,
    "languages": ["en", "fr"]
  }
}

22:39:50 in ~/tmp/26-ngx on  master 
➜ l src/main/webapp/i18n/
total 12K
drwxrwxr-x 3 pgrimaud pgrimaud 4,0K juil.  9 22:38 .
drwxrwxr-x 7 pgrimaud pgrimaud 4,0K juil.  9 22:38 ..
drwxrwxr-x 2 pgrimaud pgrimaud 4,0K juil.  9 22:38 fr

22:39:52 in ~/tmp/26-ngx on  master 
➜ 
```

**Comment by pascalgrimaud** [comment]
Another issue: when regenerating project, the prompt appears:

```
This is an existing project, using the configuration from your .yo-rc.json file 
to re-generate the project...

? Please choose additional languages to install (Press <space> to select, <a> to toggle all, <i> to invert selection)
❯◯ Albanian
 ◯ Arabic (Libya)
 ◯ Armenian
 ◯ Belarusian
 ◯ Bengali
 ◯ Bulgarian
 ◯ Catalan
```

**Comment by mshima** [comment]
> @mshima : I merged it too quickly. I found a bug when generating project with languages `en` + `fr`. Only `fr` files are generated:
> 
> ```
>     "nativeLanguage": "en",
>     "packageFolder": "io/github/pascalgrimaud/burger",
>     "jwtSecretKey": "ZmZhNjhhYjFlODRkNzZlMGFlN2FhZWRmNTkyZDdlMWZiMTlhZGI4ZWNkYTM2ZjVmMDI3N2I2NDVhMTAxZGVkM2JiNTYxYTk0YmFhNTQwYTQ0ZmY5Y2FlN2Q0Y2MzNTc5YmYwOWI3ZGM5MjJjMzM4YzRhN2M0OTU3ZTU1M2U0YWM=",
>     "clientThemeVariant": "primary",
>     "useSass": true,
>     "languages": ["en", "fr"]
>   }
> }
> 
> 22:39:50 in ~/tmp/26-ngx on  master 
> ➜ l src/main/webapp/i18n/
> total 12K
> drwxrwxr-x 3 pgrimaud pgrimaud 4,0K juil.  9 22:38 .
> drwxrwxr-x 7 pgrimaud pgrimaud 4,0K juil.  9 22:38 ..
> drwxrwxr-x 2 pgrimaud pgrimaud 4,0K juil.  9 22:38 fr
> 
> 22:39:52 in ~/tmp/26-ngx on  master 
> ➜ 
> ```

From prompt?

**Comment by mshima** [comment]
@pascalgrimaud I will fix at https://github.com/jhipster/generator-jhipster/pull/12056.
I forgot a `.only` at test case. 🤦

**Comment by pascalgrimaud** [comment]
I still use prompt a lot, yes :)
Thanks a lot @mshima

**Pull request merged**